addons/test_mail_full/tests/test_controller_common.py

```diff
@@ -0,0 +1,17 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.mail.tests.test_controller_common import TestControllerCommon
+
+
+def _get_sign_token_params(self, record):
+    access_token = record._portal_ensure_token()
+    partner = record.env["res.partner"].create({"name": "Sign Partner"})
+    _hash = record._sign_token(partner.id)
+    token = {"token": access_token}
+    bad_token = {"token": "incorrect token"}
+    sign = {"hash": _hash, "pid": partner.id}
+    bad_sign = {"hash": "incorrect hash", "pid": partner.id}
+    return token, bad_token, sign, bad_sign, partner
+
+
+setattr(TestControllerCommon, "_get_sign_token_params", _get_sign_token_params)
```

Comment:
@seb-odoo 
Do I have a better solution to add this method to the `TestControllerCommon`?

Reply:
Since it only uses the `record` and not `self`, it's possible to define it as an independent function and not add it to `TestControllerCommon`.

Reply:
I would say create `TestPortalControllerCommon` inheriting from `TestControllerCommon` (to be done in portal module test?) and also extend that class in portal related tests?

Reply:
Done